components/transfer/commons/transfer_row_display.lua

```diff
@@ -0,0 +1,366 @@
+				'This transfer was not formally announced, ' ..
+					'but was revealed by a change in the LoL Esports League-Recognized Contract Database'
+			), 'https://docs.google.com/spreadsheets/d/1Y7k5kQ2AegbuyiGwEPsa62e883FYVtHqr6UVut9RC4o/pubhtml#')
+		end
+		return nil
+	end)
+end
+
+---@return Html?
+function TransferRowDisplay:build()
+	local transfer = self.transfer
+	if Logic.isEmpty(transfer) then return end
+
+	return self
+		:cssClass()
+		:date()
+		:platform()
+		:players()
+		:from()
+		:icon()
+		:to()
+		:references()
+		:create()
+end
+
+---@return self
+function TransferRowDisplay:cssClass()
+	self.display:addClass('divRow mainpage-transfer-' .. self:_getStatus())
+	return self
+end
+
+---@return string
+function TransferRowDisplay:_getStatus()
+	local transfer = self.transfer
+	if transfer.from.teams[1] and transfer.to.teams[1] then
+		return 'neutral'
+	elseif transfer.from.teams[1] then
+		return 'from-team'
+	elseif transfer.to.teams[1] then
+		return 'to-team'
+	elseif self:_isSpecialRole(transfer.from.roles[1]) then
+		return 'to-team'
+	elseif self:_isSpecialRole(transfer.to.roles[1]) then
+		return 'from-team'
+	end
+
+	return 'neutral'
+end
+
+---@param role string?
+---@return boolean
+function TransferRowDisplay:_isSpecialRole(role)
+	if not role then return false end
+	role = role:lower()
+	return Table.includes(SPECIAL_ROLES, role)
+end
+
+---@return self
+function TransferRowDisplay:date()
+	self.display:tag('div')
+		:addClass('divCell Date')
+		:wikitext(self.transfer.date)
+	return self
+end
+
+---@return self
+function TransferRowDisplay:platform()
+	if not self.transfer.platform then return self end
+
+	self.display:tag('div')
+		:addClass('divCell GameIcon')
+		:wikitext(self.transfer.platform)
+
+	return self
+end
+
+---@return self
+function TransferRowDisplay:players()
+	local playersCell = self.display:tag('div')
+		:addClass('divCell Name')
+
+	Array.forEach(self.transfer.players, function(player, playerIndex)
+		playersCell:node(PlayerDisplay.BlockPlayer{player = player})
+	end)
+
+	return self
+end
+
+---@return self
+function TransferRowDisplay:from()
+	self.display:node(self:_displayTeam{
+		data = self.transfer.from,
+		date = self.transfer.date,
+		isOldTeam = true,
+	})
+	return self
+end
+
+---@param args {isOldTeam: boolean, date: string, data: {teams: string[], roles: string[]}}
+---@return Html
+function TransferRowDisplay:_displayTeam(args)
+	local isOldTeam = args.isOldTeam
+	local data = args.data
+	local align = isOldTeam and 'right' or 'left'
+	local teamCell = mw.html.create('div')
+		:addClass('divCell Team ' .. (isOldTeam and 'OldTeam' or 'NewTeam'))
+		:css('text-align', align)
+
+	if not data.teams[1] and not data.roles[1] then
+		return teamCell:node(self:_createRole{'&nbsp;None&nbsp;'}:css('margin-' .. align, '60px'))
+	end
+
+	local displayTeam = isOldTeam and mw.ext.TeamTemplate.team2short or mw.ext.TeamTemplate.teamshort
```

Comment:
As stated previously, IMO the display would be better with only one order
```suggestion
	local displayTeam = mw.ext.TeamTemplate.teamshort
```

Reply:
personally i am indifferent to flip or not flip